Dockerfile

```diff
@@ -0,0 +1,35 @@
+FROM python:3.8-slim
+
+# Install Python tools (git + pipenv)
+RUN apt-get update && apt-get install -y git
+RUN pip install pipenv
+
+# Install memory_profiler if this script is run with PROFILE_MEMORY flag
+ARG INSTALL_MEMORY_PROFILER="false"
+RUN if [ "$INSTALL_MEMORY_PROFILER" = "true" ]; then \
+        apt-get update && apt-get install -y gcc && \
+        pip install memory_profiler; \
+    fi
+
+# Make a directory for private credentials files
+RUN mkdir /credentials
+
+# Make a directory for intermediate data
+RUN mkdir /data
+
+# Set working directory
+WORKDIR /app
+
+# Install project dependencies.
+ADD Pipfile /app
+ADD Pipfile.lock /app
+RUN pipenv sync
+
+# Copy the rest of the project
+ADD code_schemes/*.json /app/code_schemes/
+ADD configurations/ /app/configurations/
+ADD src /app/src
+ADD sync_coda_to_engagement_db.py /app
+ADD sync_engagement_db_to_coda.py /app
+ADD sync_engagement_db_to_rapid_pro.py /app
+ADD engagement_db_to_analysis.py /app
```

Comment:
It would probably have been a good idea to dockerise one of the steps at the same time so you could test this before sending for review. I think this will fail when dockerised because we now need the .git repo to be available too, which either means copying this folder too or allowing us to pass git info via arguments

Reply:
I'll be dockerising one of the steps in a new PR.